pastas/check.py

```diff
@@ -0,0 +1,555 @@
+    "equal": "==",
+    "not_equal": "!=",
+}
+
+
+def _stat_ufunc_threshold(
+    ml: Model,
+    ufunc: Callable,
+    statistic: str,
+    threshold: float,
+):
+    """Generic function to compare a model statistic with a threshold using a ufunc.
+
+    Parameters
+    ----------
+    ml: pastas.Model
+        Pastas model instance.
+    ufunc: callable
+        Numpy ufunc (e.g. np.greater_than) to compare the model statistic with the
+        threshold.
+    statistic: str
+        Name of the statistic to be compared.
+    threshold: float
+        Threshold value for the statistic.
+
+    Returns
+    -------
+    df: pandas.DataFrame
+        DataFrame with the results of the check.
+    """
+    val = getattr(ml.stats, statistic)()
+    check = ufunc(val, threshold)
+    label = f"{statistic}{operators[ufunc.__name__]}{threshold}"
+    df = DataFrame(
+        index=[label],
+        columns=["statistic", "operator", "threshold", "unit", "pass", "comment"],
+    )
+    df.index.name = "check"
+    df.loc[label] = [val, operators[ufunc.__name__], threshold, "-", check, ""]
+    return df
+
+
+def rsq_geq_threshold(ml: Model, threshold: float = 0.7):
+    """Check R^2 >= threshold.
+
+    Parameters
+    ----------
+    ml: pastas.Model
+        Pastas model instance.
+    threshold: float
+        Threshold value for the R^2 statistic.
+
+    Returns
+    -------
+    df: pandas.DataFrame
+        DataFrame with the results of the check.
+    """
+    return _stat_ufunc_threshold(ml, np.greater_equal, "rsq", threshold)
+
+
+def response_memory(
+    ml,
+    cutoff: float = 0.95,
+    factor_length_oseries: float = 0.5,
+    stressmodel: Optional[Union[list[str], str]] = None,
+):
+    """Check if response function memory is shorter than fraction of calibration period.
+
+    Parameters
+    ----------
+    ml: pastas.Model
+        Pastas model instance.
+    cutoff: float, optional
+        Cutoff value for the response function. Default is 0.95.
+    factor_length_oseries: float, optional
+        Factor to multiply the length of the observation series with to get the
+        maximum allowed memory. Default is 0.5, e.g. half of the calibration period.
+    stressmodel: list or str, optional
+        List of stressmodel names to check the memory for. Default is None, which
+        means all stressmodels are checked.
+
+    Returns
+    -------
+    df: pandas.DataFrame
+        DataFrame with the results of the check.
+    """
+    len_oseries_calib = (ml.settings["tmax"] - ml.settings["tmin"]).days
+
+    if stressmodel is None:
+        stressmodel = list(ml.stressmodels.keys())
+    elif stressmodel is not None and not isinstance(stressmodel, list):
+        stressmodel = [stressmodel]
+
+    df = DataFrame(
+        columns=["statistic", "operator", "threshold", "unit", "pass", "comment"]
+    )
+    df.index.name = "check"
+    unit = "days"
+
+    def interp_step(cutoff: float, p: np.ndarray, rfunc: RfuncBase):
```

Comment:
For the `np.ndarray` type hint. I think you can also do; `np.ndarray[float]`. Otherwise you could also use `ps.typing.ArrayLike`.